homeassistant/components/arlo/sensor.py

```diff
@@ -51,43 +82,44 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
 
     sensors = []
     for sensor_type in config[CONF_MONITORED_CONDITIONS]:
+        sensor_entry = SensorEntityDescription(SENSOR_TYPES[sensor_type])
         if sensor_type == "total_cameras":
-            sensors.append(ArloSensor(SENSOR_TYPES[sensor_type][0], arlo, sensor_type))
+            sensors.append(ArloSensor(sensor_entry.name, arlo, sensor_entry))
         else:
             for camera in arlo.cameras:
                 if sensor_type in ("temperature", "humidity", "air_quality"):
                     continue
 
-                name = f"{SENSOR_TYPES[sensor_type][0]} {camera.name}"
-                sensors.append(ArloSensor(name, camera, sensor_type))
+                name = f"{sensor_entry.name} {camera.name}"
+                sensors.append(ArloSensor(name, camera, sensor_entry))
 
             for base_station in arlo.base_stations:
                 if (
                     sensor_type in ("temperature", "humidity", "air_quality")
                     and base_station.model_id == "ABC1000"
                 ):
-                    name = f"{SENSOR_TYPES[sensor_type][0]} {base_station.name}"
-                    sensors.append(ArloSensor(name, base_station, sensor_type))
+                    name = f"{sensor_entry.name} {base_station.name}"
+                    sensors.append(ArloSensor(name, base_station, sensor_entry))
 
     add_entities(sensors, True)
 
 
 class ArloSensor(SensorEntity):
     """An implementation of a Netgear Arlo IP sensor."""
 
-    def __init__(self, name, device, sensor_type):
+    def __init__(self, name, device, sensor_entry):
         """Initialize an Arlo sensor."""
         _LOGGER.debug("ArloSensor created for %s", name)
-        self._name = name
+        sensor_entry.name = name
+        if sensor_entry.key == "temperature":
+            sensor_entry.device_class = DEVICE_CLASS_TEMPERATURE
+        elif sensor_entry.key == "humidity":
+            sensor_entry.device_class = DEVICE_CLASS_HUMIDITY
```

Comment:
Set this in the `SensorEntityDescription` object

Reply:
Done.

Reply:
I misinterpreted what you wanted, done now.